pyuvdata/uvdata/uvh5.py

```diff
@@ -532,12 +561,16 @@ def _read_header(
             proc = self.set_lsts_from_time_array(background=background_lsts)
 
         # get frequency information
-        self.freq_array = header["freq_array"][:, :]
+        self.freq_array = header["freq_array"][:]
         self.spw_array = header["spw_array"][:]
 
+        if self.freq_array.ndim == 1:
+            assert np.asarray(header["channel_width"]).size == self.freq_array.size
+            self._set_future_array_shapes()
+
         # Pull in the channel_width parameter as either an array or as a single float,
         # depending on whether or not the data is stored with a flexible spw.
-        if self.flex_spw:
+        if self.flex_spw or np.asarray(header["channel_width"]).ndim == 1:
```

Comment:
This is admittedly bit of a corner case, but is there ever a situation where `flex_spw = True` and `np.asarray(header["channel_width"]).ndim != 1:`? I suspect the answer is no (seems like that case should throw an error), in which case I might ask if there is any advantage in using this new logic instead of the original one that just checks `flex_spw`.

Reply:
Yes, there is. We agreed that we want the uvh5 files to have the same shapes as the object in memory. So if you call `set_future_array_shapes` and then write a uvh5 file, it may have only one SPW but the channel width will be an array. In that case we read it in properly and set the `future_array_shapes` boolean on the object. In the future, of course, this will always be the case but we will have to support old files indefinitely.